nixos/modules/services/networking/crproxy.nix

```diff
@@ -0,0 +1,166 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+let
+  cfg = config.services.crproxy;
+
+  allowImageListFile = pkgs.writeTextFile {
+    name = "crproxy-allow-image-list";
+    text = lib.strings.concatLines cfg.allowImageList;
+  };
+  useAllowImageList = (lib.length cfg.allowImageList) != 0;
+  blockIPListFile = pkgs.writeTextFile {
+    name = "crproxy-block-ip-list";
+    text = lib.strings.concatLines cfg.blockIPList;
+  };
+  useBlockIPList = (lib.length cfg.blockIPList) != 0;
+  inherit (lib)
+    mkIf
+    mkEnableOption
+    mkOption
+    types
+    concatStringsSep
+    concatLists
+    optionals
+    literalExpression
+    mkPackageOption
+    ;
+in
+{
+  options.services.crproxy = {
+    enable = mkEnableOption "CRProxy (Container Registry Proxy) is a generic image proxy";
+
+    package = mkPackageOption pkgs "crproxy" { };
+
+    listenAddress = mkOption {
+      default = ":8080";
+      example = literalExpression ''
+        ":8080"
+      '';
+      type = types.str;
+      description = ''listen on the address (default ":8080")'';
+    };
+
+    behindProxy = mkOption {
+      default = false;
+      example = literalExpression ''
+        true
+      '';
+      type = types.bool;
+      description = "Behind the reverse proxy";
+    };
+
+    userpass = mkOption {
+      default = [ ];
+      example = literalExpression ''
+        [ "user@pwd@host" ]
+      '';
+      type = types.listOf types.str;
+      description = "host and username and password -u user:pwd@host";
```

Comment:
Please explain what this option should be set to. Where is the value used?

Reply:
see https://github.com/DaoCloud/crproxy/blob/e9aed8ed79a6e3435f87895fe98b91918acee464/cmd/crproxy/main.go#L84

it's a list that for some container registries need login

eg.
```nix
services.crproxy.userpass = [
  "user1:password@docker.io"
  "user2:password@ghcr.io"
];
```